src/Cassandra/Mapping/TypeConversion/TypeConverter.cs

```diff
@@ -61,6 +61,42 @@ protected TypeConverter()
             return converter(value);
         }
 
+        internal object ConvertObjectValue(Type valueType, Type dbType, object value)
+        {
+            var converter = GetToDbConverter(valueType, dbType);
+            if (converter == null)
+            {
+                try
+                {
+                    return Convert.ChangeType(value, dbType);
```

Comment:
Hum, it seems that removing that cast created a regression : List and Set still need to be cast to IEnumerable when uploaded to the db.